In [21]:
import pandas as pd

In [22]:
data=pd.read_csv("D:/ADMIN/Desktop/day27/train-1.csv")
data.head(2)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   

   Parch     Ticket     Fare Cabin Embarked  
0      0  A/5 21171   7.2500   NaN        S  
1      0   PC 17599  71.2833   C85        C

In [23]:
data.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [24]:
data=data.drop(['PassengerId','Name','Ticket', 'Cabin'],axis=1)
data.head(2)

Survived  Pclass     Sex   Age  SibSp  Parch     Fare Embarked
0         0       3    male  22.0      1      0   7.2500        S
1         1       1  female  38.0      1      0  71.2833        C

In [25]:
from sklearn import preprocessing, svm
from sklearn.metrics import confusion_matrix,accuracy_score
from sklearn.model_selection import train_test_split

In [26]:
le=preprocessing.LabelEncoder()
le.fit(data['Sex'])
data["Sex"]=le.transform(data["Sex"])
le.fit(data["Embarked"])
data["Embarked"]=le.transform(data["Embarked"])

In [27]:
DATA_ATT=[ 'Survived', 'Pclass','Sex', 'SibSp',
       'Parch', 'Embarked']
for  i in DATA_ATT:
    y = data[i]
    x=data.drop([i],axis=1)
    x_train,x_test,y_train,y_test= train_test_split(x,y,test_size=0.3,random_state=0)
    clf=svm.SVC(gamma=0.01,C=100)
    clf.fit(x_train,y_train)
    y_pred=clf.predict(x_test)
    print("accuracy score for feature {a} is {b}".format(a=i, b=accuracy_score(y_test,y_pred,normalize=True)))
    print("Confusion Matrix\n",confusion_matrix(y_test,y_pred))

accuracy score for feature Survived is 0.7415730337078652
Confusion Matrix
 [[124  33]
 [ 36  74]]
accuracy score for feature Pclass is 0.9026217228464419
Confusion Matrix
 [[ 64   4   2]
 [  4  36   9]
 [  1   6 141]]
accuracy score for feature Sex is 0.7265917602996255
Confusion Matrix
 [[ 66  32]
 [ 41 128]]
accuracy score for feature SibSp is 0.7490636704119851
Confusion Matrix
 [[150  24   7   0   1   0   0]
 [ 21  39   5   0   0   0   0]
 [  4   1   3   0   0   0   0]
 [  0   2   1   3   1   0   0]
 [  0   0   0   0   2   0   0]
 [  0   0   0   0   0   1   0]
 [  0   0   0   0   0   0   2]]
accuracy score for feature Parch is 0.797752808988764
Confusion Matrix
 [[187   6   5   0   0   0]
 [ 21  15   3   0   0   1]
 [ 10   6  11   0   0   0]
 [  0   0   1   0   0   0]
 [  1   0   0   0   0   0]
 [  0   0   0   0   0   0]]
accuracy score for feature Embarked is 0.7677902621722846
Confusion Matrix
 [[ 23   2  24]
 [  0   8  14]
 [ 18   4 174]]


#INFERENCE:
The accuracy is highest when we took the categorical dependent attribute to be "Pclass".
This model classifies 90.26% of its test data records correctly.